tests/config_change_condition/config_change_condition_suite_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright DataStax, Inc.
+// Please see the included license file for details.
+
+package config_change
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+
+	ginkgo_util "github.com/datastax/cass-operator/mage/ginkgo"
+	"github.com/datastax/cass-operator/mage/kubectl"
+)
+
+var (
+	testName         = "Config change condition"
+	namespace        = "test-config-change-condition"
+	dcName           = "dc2"
+	dcYaml           = "../testdata/default-single-rack-2-node-dc.yaml"
+	dcResource       = fmt.Sprintf("CassandraDatacenter/%s", dcName)
+	dcLabel          = fmt.Sprintf("cassandra.datastax.com/datacenter=%s", dcName)
+	ns               = ginkgo_util.NewWrapper(testName, namespace)
+)
+
+func TestLifecycle(t *testing.T) {
+	AfterSuite(func() {
+		logPath := fmt.Sprintf("%s/aftersuite", ns.LogDir)
+		kubectl.DumpAllLogs(logPath).ExecV()
+		fmt.Printf("\n\tPost-run logs dumped at: %s\n\n", logPath)
+		ns.Terminate()
+	})
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, testName)
+}
+
+var _ = Describe(testName, func() {
+	Context("when in a new cluster", func() {
+		Specify("the Updating condition is set for config updates", func() {
+			By("creating a namespace")
+			err := kubectl.CreateNamespace(namespace).ExecV()
+			Expect(err).ToNot(HaveOccurred())
+
+			step := "setting up cass-operator resources via helm chart"
+			ns.HelmInstall("../../charts/cass-operator-chart")
+
+			ns.WaitForOperatorReady()
+
+			step = "creating a datacenter resource with 1 racks/2 nodes"
+			k := kubectl.ApplyFiles(dcYaml)
+			ns.ExecAndLog(step, k)
+
+			ns.WaitForDatacenterReady(dcName)
+
+			step = "change the config"
+			json := "{\"spec\": {\"config\": {\"cassandra-yaml\": {\"file_cache_size_in_mb\": 123}, \"jvm-server-options\": {\"garbage_collector\": \"CMS\"}}}}"
+			k = kubectl.PatchMerge(dcResource, json)
+			ns.ExecAndLog(step, k)
+
+			ns.WaitForDatacenterCondition(dcName, "Updating", string(corev1.ConditionTrue))
+			ns.WaitForDatacenterCondition(dcName, "Updating", string(corev1.ConditionFalse))
+			ns.WaitForDatacenterOperatorProgress(dcName, "Ready", 1800)
+		})
+	})
+})
```

Comment:
our test framework is in great shape when we can do this test in under 100 lines! cc @sandoichi